src/hooks/useOpenLoanHandler.tsx

```diff
@@ -73,6 +76,22 @@ export const useOpenLoanHandler = () => {
     }
   }, [provider])
 
+  const getTokenBalanceAsync = async (setTokenBalance: Dispatch<SetStateAction<string>>, token: Token) => {
```

Comment:
I'm not undertstanding, why can't this function be in the utils file?

Reply:
the function is used to call `getTokenBalance` which is an async function. I think this is fine to keep here. The function name might be a little confusing

Reply:
It is because if I made this function in the utils when I am going to call this function I will need to wrap it in a async function again as useeffect does not allow await directly.